### PR TITLE
[ticket/12542] Highlight textarea when files are dragged onto editor

### PR DESCRIPTION
**Pull request by marc1706**
Ticket: https://tracker.phpbb.com/browse/PHPBB3-12542
Video: https://www.m-a-styles.de/download/drag_n_drop.mp4

Works for both posts and private messages if the attachment tab exists (attachments are enabled).

### CODE REVIEW

**Comment by nickvergessen** [comment]
@PayBas can you have a look?

**Comment by VSEphpbb** [comment]
I don't think the browser prefixes are needed for all the `outline` css properties:
http://caniuse.com/#search=outline
http://caniuse.com/#search=outline-color
https://developer.mozilla.org/en-US/docs/Web/CSS/outline-offset

**Comment by Nicofuma** [comment]
@prototech @callumacrae

**Comment by marc1706** [comment]
As a side note, IE does not support the outline-offset. The highlight border is therefore on the outside of the textarea. It does not seem to cover anything else though and still seems to serve the intended purpose.

File: phpBB/assets/javascript/core.js [hunk 1]
```diff
@@ -1233,6 +1233,31 @@ phpbb.applyCodeEditor = function(textarea) {
...
 /**
+ * Show drag and drop animation when textarea is present
+ *
+ * This function will enable the drag and drop animation for a specified
+ * textarea.
+ *
+ * @param {object} textarea Textarea DOM object to apply editor to
+ */
+phpbb.showDragNDrop = function(textarea) {
+	if (textarea === 'undefined') {
```
Comment (callumacrae) [inline phpBB/assets/javascript/core.js]: You're missing a `typeof` here, this code won't work

File: phpBB/styles/prosilver/theme/colours.css [hunk 2]
```diff
@@ -977,6 +977,22 @@ fieldset.quick-login input.inputbox {
 	color: #333333;
 }
 
+#message-box textarea.drag-n-drop {
```
Comment (callumacrae) [inline phpBB/styles/prosilver/theme/colours.css]: This seems overqualified. Would just `.drag-n-drop` not work?

**Comment by marc1706** [comment]
Updated

**Comment by callumacrae** [comment]
:+1:

Also :+1: on dropping the vendor prefixes, doesn't seem necessary at all: http://caniuse.com/#feat=outline

**Pull request merged**